pkg/state/script_caller.go

```diff
@@ -272,6 +273,112 @@ func (a *scriptCaller) invokeFunction(tree *ride.Tree, tx *proto.InvokeScriptWit
 	return true, r.ScriptActions(), err
 }
 
+func ConvertDecodedEthereumArgumentsToProtoArguments(decodedArgs []ethabi.DecodedArg) []proto.Argument {
+	var arguments []proto.Argument
+	for _, decodedArg := range decodedArgs {
+		var arg proto.Argument
+		switch decodedArg.Value.(type) {
+		case ride.RideInt:
+			arg = &proto.IntegerArgument{}
+		case ride.RideBoolean:
+			arg = &proto.BooleanArgument{}
+		case ride.RideBytes:
+			arg = &proto.BinaryArgument{}
+		case ride.RideString:
+			arg = &proto.StringArgument{}
+		case ride.RideList:
+			arg = &proto.ListArgument{}
+		}
+		arguments = append(arguments, arg)
+
+	}
+	return arguments
```

Comment:
You return list with empty arguments. `arg` in each for loop should be initialized by `decodedArg.Value`.